### PR TITLE
Dagstore Shard migration for Boost deals

### PR DESCRIPTION
**Pull request by aarshkshah1992**
- Starting legacy markets will take care of marking legacy deals for dagstore migration.

### TODO
- [x] CLI command to reinit all dagstore shards (copy over `lotus dagstore initialize-all`)
- [ ] Test on Sofia Miner.

### CODE REVIEW

File: indexprovider/wrapper.go [hunk 1]
```diff
@@ -143,3 +168,131 @@ func (w *Wrapper) AnnounceBoostDeal(ctx context.Context, pds *types.ProviderDeal
...
+			}
+		}
+	}()
+
+	// Filter for deals that are handed off.
+	//
+	// If the deal has not yet been handed off to the sealing subsystem, we
+	// don't need to call RegisterShard in this migration; RegisterShard will
+	// be called in the new code once the deal reaches the state where it's
+	// handed off to the sealing subsystem.
```
Comment (dirkmc) [inline indexprovider/wrapper.go:231]: Is this comment still relevant?

File: indexprovider/wrapper.go [hunk 2]
```diff
@@ -143,3 +168,131 @@ func (w *Wrapper) AnnounceBoostDeal(ctx context.Context, pds *types.ProviderDeal
...
+	var registered int
+	for _, deal := range deals {
+		pieceCid := deal.ClientDealProposal.Proposal.PieceCID
+
+		// enrich log statements in this iteration with deal ID and piece CID.
+		log := log.With("deal_id", deal.ChainDealID, "piece_cid", pieceCid)
+
+		// Filter for inflight deals as they will be re-indexed anyways
```
Comment (dirkmc) [inline indexprovider/wrapper.go:239]:
```suggestion
		// Filter out deals that have not yet been indexed and announced as they will be re-indexed anyways
```

File: indexprovider/wrapper.go [hunk 3]
```diff
@@ -143,3 +168,131 @@ func (w *Wrapper) AnnounceBoostDeal(ctx context.Context, pds *types.ProviderDeal
...
+		if err != nil {
+			log.Warnw("failed to register boost shard", "error", err)
+			continue
+		}
+		registered++
+	}
+
+	log.Infow("finished registering all boost shards", "total", registered)
+	totalCh <- registered
+	<-doneCh
```
Comment (dirkmc) [inline indexprovider/wrapper.go:259]: Suggest we also select against ctx.Done() here

File: indexprovider/wrapper.go (hunk 1 above)
Comment (aarshkshah1992) [reply indexprovider/wrapper.go]: removed.

**Review by dirkmc — APPROVED** [review]

**Comment by aarshkshah1992** [comment]
Tested of Mainnet and lgtm

```
2022-03-25T08:56:48.290+0200    INFO    dagstore.migrator       dagstore/wrapper.go:369 confirmed registration of all shards
2022-03-25T08:56:48.290+0200    INFO    dagstore.migrator       dagstore/wrapper.go:376 successfully marked migration as complete
2022-03-25T08:56:48.290+0200    INFO    dagstore.migrator       dagstore/wrapper.go:379 dagstore migration complete
2022-03-25T08:56:48.290+0200    INFO    storagemarket_impl      impl/provider.go:813    dagstore migration completed successfully


2022-03-25T08:56:48.378+0200    INFO    index-provider-wrapper.boost-migrator   indexprovider/wrapper.go:219    async boost shard registration completed successfully   {"shard_key": "baga6ea4seaqcuivce6if3vfazpl7oa4bwj2ioxf4p4wxwwbzon2oc7slqq6ckoi"}
2022-03-25T08:56:48.378+0200    INFO    index-provider-wrapper.boost-migrator   indexprovider/wrapper.go:260    confirmed registration of all boost shards
2022-03-25T08:56:48.378+0200    INFO    index-provider-wrapper.boost-migrator   indexprovider/wrapper.go:267    successfully marked boost migration as complete
2022-03-25T08:56:48.378+0200    INFO    index-provider-wrapper.boost-migrator   indexprovider/wrapper.go:270    boost dagstore migration complete



(39/41) baga6ea4seaqd6bqsljx7dsvf3zyaaaiu2u6mom5ikh3kp3a44two3jbaaa3kioq STARTING
(38/41) baga6ea4seaqctrm7lglox53itoj3u36deyos5c2xg3pkgl32axa5iazrs75p2fa SUCCESS
(40/41) baga6ea4seaqo4b55hvq4ake5o2sw4mf6ukyppiqw2qfmltqnbs3tr4cbjcyhiiq STARTING
(39/41) baga6ea4seaqd6bqsljx7dsvf3zyaaaiu2u6mom5ikh3kp3a44two3jbaaa3kioq SUCCESS
(41/41) baga6ea4seaqctmgz3sbd3wdk7tnszh6pfcqxujjdhn225uouyuvtl3yoqywfwky STARTING
(41/41) baga6ea4seaqctmgz3sbd3wdk7tnszh6pfcqxujjdhn225uouyuvtl3yoqywfwky SUCCESS
(40/41) baga6ea4seaqo4b55hvq4ake5o2sw4mf6ukyppiqw2qfmltqnbs3tr4cbjcyhiiq SUCCESS
```

**Pull request merged**